### PR TITLE
Feature: Allow functions to be added to collection script domain

### PR DESCRIPTION
**Pull request by wuurrd**
Often one does repetitive things in the script domain, e.g. create a unique validator that one wishes to reuse, or a simple function that needs to be done on almost all collections, so this commit allows the users of deployd create generic functions they can inject into the script domain.

### CODE REVIEW

**Comment by ericfong** [comment]
Any update on this?

**Comment by wuurrd** [comment]
@ericfong Sorry for the delay, hadn't checked my notifications for a while.

Added @luto 's fix, and made CI green.

**Comment by andreialecu** [comment]
It looks like this can also be used for other things (objects, arrays, etc), not just functions. A possible suggestion for a better name could be `Collection.ExtendDomain()`. (I would also Pascal case it now that it is static)

File: lib/resources/collection/index.js [hunk 1]
```diff
@@ -596,6 +602,7 @@ function createDomain(data, errors) {
     'this': data,
     data: data
   };
+  util._extend(domain, Collection.domainAdditions);
```
Comment (ericfong) [inline lib/resources/collection/index.js]: shall we use underscore extend instead of non-public api from node util ?

**Comment by andreialecu** [comment]
Closing in favor of https://github.com/deployd/deployd/pull/593

**Comment by luto** [comment]
Thank you for taking care of this!

**Pull request closed without merging**